test/components/record/RecordForm_test.jsx

```diff
@@ -0,0 +1,139 @@
+    collection: {
+      data: {
+        ...defaultProps.collection.data,
+        attachment: {
+          enabled: true,
+          required: true
+        }
+      } 
+    }
+  };
+
+  it("Renders an empty form for a new record (attachments disabled)", async () => {
+    const result = renderWithProvider(<RecordForm {...defaultProps} />);
+    fireEvent.change(result.queryByLabelText("TestTitle"), { target: { value: "test title" }});
+    fireEvent.change(result.queryByLabelText("TestContent"), { target: { value: "test content" }});
+    fireEvent.click(result.queryByText("Create record"));
+    expect(lastSubmittedData).toStrictEqual({
+      title: "test title",
+      content: "test content"
+    });
+  });
+
+  it("Renders the expected form for an existing record (attachments disabled)", async () => {
+    const result = renderWithProvider(<RecordForm {...defaultProps} record={{
+      data: {
+        title: "test title",
+        content: "test content",
+      }
+    }} />);
+    expect(result.queryByLabelText("TestTitle").value).toBe("test title");
+    expect(result.queryByLabelText("TestContent").value).toBe("test content");
+  });
+
+  it("Renders an empty form for a new record (attachments enabled)", async () => {
+    const result = renderWithProvider(<RecordForm {...attachProps} />);
+    expect(result.queryByLabelText("TestTitle").value).toBe("");
+    expect(result.queryByLabelText("TestContent").value).toBe("");
+    expect(result.findByLabelText("File attachment")).toBeDefined();
+  });
+
+  it("Renders the expected form for an existing record (attachments enabled)", async () => {
+    const result = renderWithProvider(<RecordForm {...attachProps} record={{
+      data: {
+        title: "test title",
+        content: "test content",
+      }
+    }} />);
+    expect(result.queryByLabelText("TestTitle").value).toBe("test title");
+    expect(result.queryByLabelText("TestContent").value).toBe("test content");
+    expect(result.findByLabelText("File attachment")).toBeDefined();
+  });
+
+  it.only("Requires an attachment to submit when it's a new record and attachments are required", async () => {
+    lastSubmittedData = null;
+    const result = renderWithProvider(<RecordForm {...attachProps} />);
+    fireEvent.change(result.queryByLabelText("TestTitle"), { target: { value: "test title" }});
+    fireEvent.change(result.queryByLabelText("TestContent"), { target: { value: "test content" }});
+    fireEvent.click(result.queryByText("Create record"));
+    expect(lastSubmittedData).toBeNull();
+  });
+
+  it("Disables the form when user cannot edit", async () => {
+    canEditRecord.mockReturnValue(false);
+    const result = renderWithProvider(<RecordForm {...defaultProps} record={{
+      data: {
+        title: "test title",
+        content: "test content",
+      }
+    }} />);
+    expect(result.queryByLabelText("TestTitle").disabled).toBe(true);
+    expect(result.queryByLabelText("TestContent").disabled).toBe(true);
```

Comment:
No expectation about file form?

Reply:
In that scenario we weren't rendering the file form, but I can switch it so we are and check it too.